### PR TITLE
Add initial sbtc stacks coordinator README.md

### PR DESCRIPTION
**Pull request by jferrant**

### CODE REVIEW

File: stacks-coordinator-mini/README.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    Signer1->>Signer1: Combine Partial Signatures 1..N
+    SignerN->>SignerN: Combine Partial Signatures 1..N
+    Signer1->>Signer1: Sign Transaction
+    SignerN->>SignerN: Sign Transaction
+
+    Note over Signer1, Blockchain: Broadcast Transaction
+    Signer1->>Blockchain: Broadcast Signed Transaction
+    Blockchain-->>Signer1: Transaction Confirmation
+    SignerN->>Blockchain: Broadcast Signed Transaction
+    Blockchain-->>SignerN: Transaction Confirmation
```
Comment (netrome) [inline stacks-coordinator-mini/README.md:37]: Isn't it the coordinator which broadcasts the signed transaction? Once signatures are aggregated the transaction only needs to be signed and broadcasted once. I would assume that's where the coordinator takes over.

i.e.
Signers ->  produce signature shares
Coordinator -> aggregate shares into final signature and broadcast tx

File: stacks-coordinator-mini/README.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+end
+```
+
+This diagram shows the sequence of events that occur when a client activates an autosigner. The AutoSigner API retrieves pending transactions from the blockchain and loops through each transaction. For each transaction, it asks the Signer to determine whether to approve or reject it based on the signer's configuration. If the Signer can determine the decision, it signs the transaction and the AutoSigner API broadcasts it to the blockchain. If the Signer cannot determine the decision, the AutoSigner API notifies the client UI about the decision failure.
```
Comment (netrome) [inline stacks-coordinator-mini/README.md:41]: Feature request: If you have time to extend this doc, I'd love to see an overview of which the main components of the signer are in the beginning of this document, which introduces the main relations between a signer, the auto signer API, the signer API, the coordinator, the sBTC contract and the blockchains. I imagine a simple graph with arrows explaining their relationships, like this:

```mermaid
graph
    AU[AutoSigner]
    CL[Client UI]
    CO[Coordinator]
    SI[Signer]
    SB[Stacks Blockchain]

    CL[Client UI]-->|Configure|AU
    CL-->|Manual signing|SI
    AU-->|Automatic signing|SI
    CO-->|Request signatures|SI
    AU-->|Fetch pending transactions|SB
```

Also, I guess it's not the API that retrieves the pending transactions but rather some process which is configurable through the AutoSigner API right?

Comment (netrome) [reply stacks-coordinator-mini/README.md:41]: Right, I see it's the signer process in the next chart.

**Review by netrome — APPROVED** [review]
Great stuff! Just one clarification question and a feature request for a leading high-level introduction of the entities referred to in the sequence diagrams from me.

File: stacks-coordinator-mini/README.md (hunk 1 above)
Comment (jferrant) [reply stacks-coordinator-mini/README.md:37]: Responding for tracking purposes. 

As discussed in the Signer WG Meeting, we are exploring the option of removing a specific coordinator and having every participating signer calculate the aggregate signature/broadcasting the final transaction (leaving the mempool to remove duplicates) to improve robustness/simplify logic.

File: stacks-coordinator-mini/README.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+participant SignerN
+participant SmartContract
+participant Blockchain
+
+Note over Signer1, Blockchain: Key Generation
+Signer1->>SmartContract: Register Public Key 1
+SignerN->>SmartContract: Register Public Key N
+
+Note over Signer1, Blockchain: Retrieve Transactions
+Signer1->>Blockchain: Retrieve Unsigned Transactions
```
Comment (jo-tm) [inline stacks-coordinator-mini/README.md:15]: something on this line...
Transactions -> Unfulfilled Deposit/Withdrawal Requests 
Unsigned Transactions -> Unfulfilled Deposit/Withdrawal Requests

File: stacks-coordinator-mini/README.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+```mermaid
+sequenceDiagram
+participant Signer1
+participant SignerN
+participant SmartContract
+participant Blockchain
+
+Note over Signer1, Blockchain: Key Generation
+Signer1->>SmartContract: Register Public Key 1
```
Comment (xoloki) [inline stacks-coordinator-mini/README.md:11]: How are the signers given permission to register their keys in the contract?  In the alpha contract the coordinator is the only entity allowed to set public keys.  Does mini actually have a way for signers to set their keys?

File: stacks-coordinator-mini/README.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+Note over Signer1, Blockchain: Retrieve Transactions
+Signer1->>Blockchain: Retrieve Unsigned Transactions
+Blockchain-->>Signer1: Unsigned Transactions
+SignerN->>Blockchain: Retrieve Unsigned Transactions
+Blockchain-->>SignerN: Unsigned Transactions
+
+Note over Signer1, Blockchain: Signing Transactions
+loop for each Pending Transaction
+    Signer1->>Signer1: Create Partial Signature 1
```
Comment (xoloki) [inline stacks-coordinator-mini/README.md:22]: How does each signer create a partial signature?  You can't create a signature share without knowing who all the other signers are.  Are you assuming everyone will sign?

File: stacks-coordinator-mini/README.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    SignerN->>SignerN: Create Partial Signature N
+    Signer1->>SignerN: Share Partial Signature 1
+    SignerN->>Signer1: Share Partial Signature N
+    Signer1->>SmartContract: Check Partial Signature N
```
Comment (xoloki) [inline stacks-coordinator-mini/README.md:26]: How does the contract check the signatures?

**Review by xoloki — COMMENTED** [review]
I asked some questions, can't really approve or request changes without answers :P

File: stacks-coordinator-mini/README.md (hunk 4 above)
Comment (jferrant) [reply stacks-coordinator-mini/README.md:11]: was approaching this as a "register my key" in the contract as "available to sign" but that it was up to the smart contract whether to accept it into the list of signers for a given cycle. Not sure how we are planning to have the logic done for determining which signers' keys are accepted.

File: stacks-coordinator-mini/README.md (hunk 5 above)
Comment (jferrant) [reply stacks-coordinator-mini/README.md:22]: The signers must be able to retrieve the list of signers in the signing round from the smart contract

File: stacks-coordinator-mini/README.md (hunk 6 above)
Comment (jferrant) [reply stacks-coordinator-mini/README.md:26]: By check I mean check that this partial share was from a public key in the registered list of keys allocated for signing.

**Pull request merged**